network.go

```diff
@@ -622,33 +632,41 @@ func (n *network) Delete() error {
 		return &UnknownNetworkError{name: name, id: id}
 	}
 
-	numEps := n.getEpCnt().EndpointCnt()
-	if numEps != 0 {
+	if !force && n.getEpCnt().EndpointCnt() != 0 {
 		return &ActiveEndpointsError{name: n.name, id: n.id}
 	}
 
-	if err = n.deleteNetwork(); err != nil {
-		return err
+	// Mark the network for deletion
+	if !n.inDelete && n.Type() != "host" && n.Type() != "null" {
```

Comment:
Is there a reason to check for the `host` and `null` type here ?
These are special types and the deletion is prevented from docker side. Why do we need special protection now ?